R/summariseClinicalRecords.R

```diff
@@ -44,66 +46,86 @@
-  assertLogical(sourceVocabulary, length = 1)
-  assertLogical(domainId, length = 1)
-  assertLogical(typeConcept, length = 1)
+  omopgenerics::assertLogical(inObservation, length = 1)
+  omopgenerics::assertLogical(standardConcept, length = 1)
+  omopgenerics::assertLogical(sourceVocabulary, length = 1)
+  omopgenerics::assertLogical(domainId, length = 1)
+  omopgenerics::assertLogical(typeConcept, length = 1)
+
+  result <- purrr::map(omopTableName,
+                       function(x) {
+                         if(omopgenerics::isTableEmpty(cdm[[x]])) {
+                           cli::cli_warn(paste0(x, " omop table is empty. Returning an empty summarised omop table."))
+                           return(omopgenerics::emptySummarisedResult())
+                         }
+                         summariseClinicalRecord(x,
+                                                 cdm = cdm,
+                                                 recordsPerPerson = recordsPerPerson,
+                                                 inObservation = inObservation,
+                                                 standardConcept = standardConcept,
+                                                 sourceVocabulary = sourceVocabulary,
+                                                 domainId = domainId,
+                                                 typeConcept = typeConcept)
+                       }
+                       ) |>
+    dplyr::bind_rows()
+
+  return(result)
+}
+
+#' @noRd
+summariseClinicalRecord <- function(omopTableName, cdm, recordsPerPerson,
+                                    inObservation, standardConcept,
+                                    sourceVocabulary, domainId, typeConcept,
+                                    call = parent.frame(3)) {
 
-  if ("observation_period" == omopgenerics::tableName(omopTable)) {
+  omopgenerics::assertClass(cdm[[omopTableName]], "omop_table", call = call)
+  omopTable <- cdm[[omopTableName]] |> dplyr::ungroup()
+
+  if ("observation_period" == omopTableName) {
     if(standardConcept){
       if(!missing(standardConcept)){
-        cli::cli_warn("standardConcept turned to FALSE, as omopTable provided is observation_period")
+        cli::cli_inform("standardConcept turned to FALSE, as observation_period is in omopTableName", call = call)
       }
       standardConcept <- FALSE
     }
     if(sourceVocabulary){
       if(!missing(sourceVocabulary)){
-        cli::cli_warn("sourceVocabulary turned to FALSE, as omopTable provided is observation_period")
+        cli::cli_inform("sourceVocabulary turned to FALSE, as observation_period is in omopTableName", call = call)
       }
       sourceVocabulary <- FALSE
     }
     if(domainId){
       if(!missing(domainId)){
-        cli::cli_warn("domainId turned to FALSE, as omopTable provided is observation_period")
+        cli::cli_inform("domainId turned to FALSE, as observation_period is in omopTableName", call = call)
```

Comment:
No. Only turned to FALSE for observation_period, as this is called separately for each table. So the package would return results with domainId TRUE for all tables for which it can be true, and FALSE for those for which it cannot (if the user specifies domainId = TRUE in the call).